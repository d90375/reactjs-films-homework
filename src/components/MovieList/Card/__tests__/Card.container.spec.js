import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import CardContainer from "../Card.container";

describe("CardContainer component", () => {
  let component;

  const mockStore = configureStore([thunk]);

  const store = mockStore({
    mock: {
      mock: "mock"
    }
  });

  const setUp = (props) => {
    return shallow(
      <Provider store={store}>
        <CardContainer {...props} />
      </Provider>
    );
  };

  beforeEach(() => {
    component = setUp();
  });

  const mockProps = {
    cardItem: { cardItem: "cardItem" },
    genresItems: {
      genresItems: "genresItems"
    },
    cardIndex: 0
  };

  it("should render connected component Card", () => {
    expect(component.find(CardContainer).length).toEqual(1);
  });

  it("should check props matches with store", () => {
    expect(component.find(CardContainer).props.output).toEqual(store.output);
  });

  it("should render Card component without props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render Card component with props", () => {
    component = setUp({ ...mockProps });
    expect(component).toMatchSnapshot();
  });
});
