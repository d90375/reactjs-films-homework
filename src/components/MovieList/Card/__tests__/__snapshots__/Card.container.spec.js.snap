// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer component should render Card component with props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <CardContainer
    cardIndex={0}
    cardItem={
      Object {
        "cardItem": "cardItem",
      }
    }
    genresItems={
      Object {
        "genresItems": "genresItems",
      }
    }
  />
</ContextProvider>
`;

exports[`CardContainer component should render Card component without props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <CardContainer
    cardIndex={0}
    cardItem={
      Object {
        "adult": false,
        "backdrop_path": "",
        "genre_ids": Array [
          0,
          0,
        ],
        "id": 0,
        "original_language": "",
        "original_title": "",
        "overview": "",
        "popularity": 0,
        "poster_path": "",
        "release_date": "",
        "title": "",
        "video": false,
        "vote_average": 0,
        "vote_count": 0,
      }
    }
    genresItems={
      Object {
        "0": "",
      }
    }
  />
</ContextProvider>
`;
